server/routerlicious/packages/services-core/src/runWithRetry.ts

```diff
@@ -73,3 +76,80 @@ export async function runWithRetry<T>(
 
     return result;
 }
+
+/**
+ * Executes a given request action while providing support for retrying on failures and taking action on error.
+ * @summary
+ * The difference between {@link requestWithRetry} and {@link runWithRetry} is that {@link runWithRetry} allows the user
+ * to decide whether to ignore the error or not, on a "fire and forget" fashion. That makes the return type of
+ * {@link runWithRetry} be slightly different, as `undefined` is a possible return value. That is not the case for
+ * {@link requestWithRetry}, which focuses on requests/operations where the user always wants the error to be
+ * bubbled up, e.g. network requests. It allows for a simpler return type, `T`, since the function would never return
+ * anything other than a `T` value - the only other possibility is a promise rejection.
+ * @param  {()=>Promise<T>} request - function to run and retry in case of error
+ * @param  {string} callName - name of the api function we are calling
+ * @param  {number} maxRetries - maximum retries after which error is thrown. Retry infinitely if set to -1
+ * @param  {number} retryAfterMs - interval factor to wait before retrying. Param to calculateIntervalMs
+ * @param  {(error)=>boolean} shouldRetry? - function that takes error and decides whether to retry on it
+ * @param  {(error, numRetries, retryAfterInterval)=>number} calculateIntervalMs
+ * function which calculates interval to wait before retrying based on error, retryAfterMs and retries so far
+ * @param  {(error)=>void} onErrorFn? - function allowing caller to define custom logic to run on error e.g. custom logs
+ */
+ export async function requestWithRetry<T>(
+    request: () => Promise<T>,
+    callName: string,
+    maxRetries: number,
+    retryAfterMs: number,
+    shouldRetry?: (error) => boolean,
```

Comment:
Similarly, it looks like all current usecases use `shouldRetryNetworkError` for this. Can we just set that as the default?

Reply:
Makes sense.